plugins/inputs/procstat/nativeFinder_windows.go

```diff
@@ -0,0 +1,109 @@
+// +build windows
+
+package procstat
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/StackExchange/wmi"
+	"github.com/shirou/gopsutil/process"
+)
+
+//Timeout is the timeout used when making wmi calls
+var Timeout = 5 * time.Second
+
+func init() {
+	wmi.DefaultClient.AllowMissingFields = true
+}
+
+type queryType string
+
+const (
+	like     = queryType("LIKE")
+	equals   = queryType("=")
+	notEqual = queryType("!=")
+)
+
+// //Pattern matches the process name on windows and will find a pattern using a WMI like query
+// func (pg *NativeFinder) Pattern(pattern string) ([]PID, error) {
+// 	var pids []PID
+// 	procs, err := getWin32ProcsByVariable("Name", like, pattern, Timeout)
+// 	if err != nil {
+// 		return pids, err
+// 	}
+// 	for _, p := range procs {
+// 		pids = append(pids, PID(p.ProcessID))
+// 	}
+// 	return pids, nil
+// }
+
+//Pattern matches on the process name
+func (pg *NativeFinder) Pattern(pattern string) ([]PID, error) {
+	var pids []PID
+	regxPattern, err := regexp.Compile(pattern)
+	if err != nil {
+		return pids, err
+	}
+	procs, err := process.Processes()
+	if err != nil {
+		return pids, err
+	}
+	for _, p := range procs {
+		name, err := p.Name()
+		if err != nil {
+			//skip, this can be caused by the pid no longer existing
+			//or you having no permissions to access it
+			continue
+		}
+		if regxPattern.MatchString(name) {
+			pids = append(pids, PID(p.Pid))
+		}
+	}
+	return pids, err
+}
+
+//FullPattern matches the cmdLine on windows and will find a pattern using a WMI like query
+func (pg *NativeFinder) FullPattern(pattern string) ([]PID, error) {
+	var pids []PID
+	procs, err := getWin32ProcsByVariable("CommandLine", like, pattern, Timeout)
+	if err != nil {
+		return pids, err
+	}
+	for _, p := range procs {
+		pids = append(pids, PID(p.ProcessID))
+	}
+	return pids, nil
+}
+
+//GetWin32ProcsByVariable allows you to query any variable with a like query
+func getWin32ProcsByVariable(variable string, qType queryType, value string, timeout time.Duration) ([]process.Win32_Process, error) {
+	var dst []process.Win32_Process
+	var query string
+	query = fmt.Sprint("WHERE ", variable, " ", qType, " \"", value, "\"")
```

Comment:
Use fmt.Sprintf for readability.  I'm nervous about escaping here, `variable` and `value` need to be escaped more rigorously.

Reply:
I changed to sprintf, how would you like me to protect against escaping? based on [this](https://codereview.stackexchange.com/questions/160021/escaping-characters-in-a-wmi-query) I should be able to just escape backslash and single quote for the value. On the variable I can just fail anytime any of the non standard characters are in it. What do you think?

Reply:
In this case you can probably use the `%q` pattern, it will escape double quote and backslash using backslash, and it will add the outer quotation marks:
```
"WHERE %s %s %q"
```